src/main/scala/akka/persistence/cassandra/query/EventsByTagPublisher.scala

```diff
@@ -269,9 +272,9 @@ private[query] class EventsByTagPublisher(
 
       deliverBuf()
 
-      if (mightBeMore) {
+      if (mightBeMore && !isLookingForDelayedEnd()) {
         self ! Continue // fetched limit, more to fetch
-      } else if (isTimeBucketBeforeToday()) {
+      } else if (isTimeBucketBeforeToday() && !isLookingForDelayedEnd()) {
         nextTimeBucket()
         self ! Continue // fetch more from next time bucket
       } else {
```

Comment:
@TrustNoOne I found what is wrong. The LookingForDelayed queries are bumbing the currentTimeBucket forward, so when the ordinary queries continue they skip the buckets at the end.

Revert above changes and add the following here instead:
```scala
if (backtrackingMode != NoBacktracking)
  currTimeBucket = TimeBucket(currOffset)
```

I'm also looking at if the LookingForDelayed can be skipped altogether before initial queries have reached today's date. That would only be an optimization.